extensions/positron-run-app/src/api.ts

```diff
@@ -389,249 +405,128 @@ export class PositronRunAppApiImpl implements PositronRunApp, vscode.Disposable
+		const localBaseUri = vscode.Uri.parse(url.toString());
+
+		// Example: http://localhost:8500/url/path or http://localhost:8500
+		const localUri = options.urlPath ?
+			vscode.Uri.joinPath(localBaseUri, options.urlPath) : localBaseUri;
+
+		// Example: http://localhost:8080/proxy/5678/url/path or http://localhost:8080/proxy/5678
+		let previewUri = undefined;
+		if (options.proxyInfo) {
+			// On Web (specifically Positron Server Web and not PWB), we need to set up the proxy with
+			// the urlPath appended to avoid issues where the app does not set the base url of the app
+			// or the base url of referenced assets correctly.
+			const applyWebPatch = IS_POSITRON_WEB && !IS_RUNNING_ON_PWB;
+			const targetOrigin = applyWebPatch ? localUri.toString(true) : localBaseUri.toString();
+
+			log.debug(`Finishing proxy setup for app at ${targetOrigin}`);
+
+			// Finish the Positron proxy setup so that proxy middleware is hooked up.
+			await options.proxyInfo.finishProxySetup(targetOrigin);
+			previewUri = !applyWebPatch && options.urlPath
+				? vscode.Uri.joinPath(options.proxyInfo.externalUri, options.urlPath)
+				: options.proxyInfo.externalUri;
+		} else {
+			previewUri = await vscode.env.asExternalUri(localUri);
+		}
 
-async function showEnableShellIntegrationMessage(rerunApplicationCallback: () => any): Promise<void> {
-	// Don't show if the user disabled this message.
-	if (!vscode.workspace.getConfiguration().get<boolean>(Config.ShowEnableShellIntegrationMessage)) {
-		return;
-	}
+		log.debug(`Viewing app at local uri: ${localUri.toString(true)} with external uri ${previewUri.toString(true)}`);
 
-	// Prompt the user to enable shell integration.
-	const enableShellIntegration = vscode.l10n.t('Enable Shell Integration');
-	const notNow = vscode.l10n.t('Not Now');
-	const dontAskAgain = vscode.l10n.t('Don\'t Ask Again');
-	const selection = await vscode.window.showInformationMessage(
-		vscode.l10n.t(
-			'Shell integration is disabled. Would you like to enable shell integration for this ' +
-			'workspace to automatically preview your application in the Viewer pane?',
-		),
-		enableShellIntegration,
-		notNow,
-		dontAskAgain,
-	);
-
-	if (selection === enableShellIntegration) {
-		// Enable shell integration.
-		const shellIntegrationConfig = vscode.workspace.getConfiguration('terminal.integrated.shellIntegration');
-		await shellIntegrationConfig.update('enabled', true, vscode.ConfigurationTarget.Workspace);
-
-		// Prompt the user to rerun the application.
-		const rerunApplication = vscode.l10n.t('Rerun Application');
-		const notNow = vscode.l10n.t('Not Now');
-		const selection = await vscode.window.showInformationMessage(
-			vscode.l10n.t('Shell integration is now enabled. Would you like to rerun the application?'),
-			rerunApplication,
-			notNow,
+		// Record the known app server URL to proxy server mapping.
+		this.addAppServer(
+			localBaseUri.toString(),
+			options.terminalPid,
+			previewUri,
```

Comment:
this is the only real change to this `previewUrlInExecutionOutput` method. The rest of the diff is from moving `previewUrlInExecutionOutput` from a standalone function in this file to a method within the `PositronRunAppApiImpl` class